### PR TITLE
Issues: Show link to conflicts doc if there are some

### PR DESCRIPTION
**Pull request by ckamm**
The link url is themed.

For #6396

Only merge once we can check that the link will actually work. Currently it's a 404: https://doc.owncloud.org/desktop/2.5/conflicts.html

### CODE REVIEW

File: src/libsync/owncloudtheme.cpp [hunk 1]
```diff
@@ -56,5 +56,11 @@ QPixmap ownCloudTheme::wizardHeaderLogo() const
     return QPixmap(hidpiFileName(":/client/theme/colored/wizard_logo.png"));
 }
 
+QString ownCloudTheme::conflictDocumentationLink() const
+{
+    return QString("https://doc.owncloud.org/desktop/%1.%2/conflicts.html")
```
Comment (ogoffart) [inline src/libsync/owncloudtheme.cpp]: Please Use Theme::helpUrl as a base,  and move the function to Theme  and remove the override in ownCloudTheme.

Comment (ckamm) [reply src/libsync/owncloudtheme.cpp]: Oh, I didn't see `helpUrl()`. Yes, I didn't want oc-branded urls to show up in themes that weren't adjusted for this change. But if `helpUrl` already exists I'll just use that.

File: src/gui/issueswidget.cpp [hunk 2]
```diff
@@ -277,6 +282,8 @@ void IssuesWidget::slotProgressInfo(const QString &folder, const ProgressInfo &p
             }
         }
         emit ProgressDispatcher::instance()->folderConflicts(folder, conflicts);
+
+        _ui->_conflictHelp->setHidden(Theme::instance()->configFileName().isEmpty() || conflicts.isEmpty());
```
Comment (ckamm) [inline src/gui/issueswidget.cpp]: oops

**Comment by ckamm** [comment]
Ok, ready to merge once help url becomes valid.

**Comment by guruz** [comment]
FYI @michaelstingl for themeing / white label stuff.

**Comment by michaelstingl** [comment]
@guruz What will change for branded clients? Could you elaborate?

**Comment by ckamm** [comment]
@michaelstingl See the screenshot in https://github.com/owncloud/client/issues/6396#issuecomment-381499838

The url that the help link points to is based on `Theme::helpUrl` by default and independently themable by overriding `Theme::conflictHelpUrl`. If a theme doesn't override either, the link will point to the ownCloud documentation, similar to the change done in 42b254785e8b91e5e2e4d8c9dfa7681d947f7f8f

**Comment by michaelstingl** [comment]
Should be brandable, similar to Help URL:

> **Help URL**
> Select this option to show a help URL in your client. The help option will be shown on menu and clicking on it will open a browser viewing the selected URL.
> - [ ] Show help url?

> **Conflict Help URL**
> Select this option to show a help URL for conflict files in your client. The help option will be shown…
> - [ ] Show conflict help url?

If this isn't set, nothing should be shown. If it's set, a new input field should be shown to enter the URL.

File: src/libsync/theme.cpp [hunk 3]
```diff
@@ -228,6 +228,11 @@ QString Theme::helpUrl() const
     return QString::fromLatin1("https://doc.owncloud.org/desktop/%1.%2/").arg(MIRALL_VERSION_MAJOR).arg(MIRALL_VERSION_MINOR);
 }
 
+QString Theme::conflictHelpUrl() const
+{
+    return helpUrl() + QStringLiteral("conflicts.html");
```
Comment (ogoffart) [inline src/libsync/theme.cpp]: When helpUrl is empty, this should be empty too.

Comment (ckamm) [reply src/libsync/theme.cpp]: updated

**Comment by ckamm** [comment]
@michaelstingl Yes, this kind of option setup would work. If the options are not checked, the theme functions are overridden to return an empty string, and if options are checked they are overridden to return the specified url.

**Comment by guruz** [comment]
https://doc.owncloud.org/desktop/2.5/conflicts.html is still 404. 
I think someone mentioned that we shall create a 2.5 branch then this can be done. So merge this (to master) so we can branch off 2.5 and then ping @tboerger @settermjd ?

**Comment by ckamm** [comment]
@guruz that was also my understanding

**Comment by ogoffart** [comment]
```
return QString::fromLatin1("https://doc.owncloud.org/desktop/%1.%2/").arg(MIRALL_VERSION_MAJOR).arg(MIRALL_VERSION_MINOR);
```

This will always be a 404 before the branching is done.  (i mean, once we change the master version to 2.6, the master dailly build will still have a 404.

**Comment by ckamm** [comment]
Yes, let's just fix the link if it ends up incorrect.

**Pull request merged**